include/trace/events/error_report.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
```

Comment:
Commit-message: it's not just restricted to "memory tools", it's more generic. I also wouldn't mention KCSAN here, because it raises the expectation that we'll add support in this series, which I think we shouldn't for now (I'll do that at some point, but not sure yet).

I think we might actually get some value out of adding the tracepoints to UBSAN as well, because UBSAN can be enabled in prod. So maybe the list of examples should be "KASAN, KFENCE, and UBSAN".

The description should also motivate better why this is useful standalone, without even considering error_report_notify. I would mention that it allows tests to hook into arbitrary bug and error reporting. And it also allows to benefit from ftrace event tracing.

Reply:
How about this:
```
Introduce error_report_start and error_report_end tracepoints.
Those can be used in debugging tools like KASAN, KFENCE or UBSAN
to provide extensions to the error reporting mechanisms (e.g. allow
tests hook into error reporting, ease error report collection from
production kernels).
Another benefit would be making use of ftrace for debugging or
benchmarking the tools themselves.
```
?

>I also wouldn't mention KCSAN here, because it raises the expectation that we'll add support in this series, which I think we shouldn't for now (I'll do that at some point, but not sure yet).

Given the new updates regarding KFENCE, shall I mention it? Or maybe we'd better send the error_report series independently and then send a single followup patch enabling traces in KFENCE?

Reply:
Sounds good, thanks.


I think we can send all as 1 series. Since this will only be included in the next cycle anyway, we can mention KFENCE. I assume this will also be picked up by -mm, so having the patch enabling traces in KFENCE is fine.

Reply:
Maybe do a s/KASAN, KFENCE or UBSAN/KASAN, KFENCE, *SAN, etc./

because I worry that somebody might call out we're not adding UBSAN support in the series. Trying to manage people's expectations and ensure we don't give them too many reasons to complain...

Unless you feel like adding UBSAN support in this series.

Reply:
I would just write "KASAN, KFENCE, etc." then, as *SAN already includes KASAN, and "*SAN and KFENCE" is a little too short.

Reply:
LGTM